### PR TITLE
gcc7 compilation warning(s) fix(es) in CommonTools/Utils

### PR DESCRIPTION
**Pull request by mrodozov**
Fix for compilation warning(s) when compiling with gcc7 and more flags listed here:
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc700/CMSSW_9_3_X_2017-07-24-2300/CommonTools/Utils

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by slava77** [comment]
@mrodozov 
please edit the PR title to be grammatically correct instead of the current stream of characters

**Comment by mrodozov** [comment]
@slava77 yes sure I've noticed I'm using the branch name as commit message and I'll change the titles with more appropriate

File: CommonTools/Utils/interface/ExpressionEvaluatorTemplates.h [hunk 1]
```diff
@@ -23,8 +24,9 @@ namespace reco {
   template<typename Object>
   struct ValueOnObject {
     virtual double eval(Object const&) const =0;
+    virtual ~ValueOnObject(){};
```
Comment (slava77) [inline CommonTools/Utils/interface/ExpressionEvaluatorTemplates.h]: in this file I also see CutOnObject and SelectIndecesInCollection with virtual methods and no virtual destructor defined.
Why?
Are these not used anywhere yet to trigger the warning?

It looks like these other types should have a destructor defined as well.

Comment (smuzaffar) [inline CommonTools/Utils/interface/ExpressionEvaluatorTemplates.h]: glimps shows that these classes are not use (at least in cmssw).
```
>cmsglimpse CutOnObject
src/CommonTools/Utils/interface/ExpressionEvaluatorTemplates.h:   struct CutOnObject {
>cmsglimpse SelectIndecesInCollection
src/CommonTools/Utils/interface/ExpressionEvaluatorTemplates.h:   struct SelectIndecesInCollection {
```
@mrodozov , I would suggest to just add the missing destructor.

**Comment by davidlange6** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/19898	d3c563d
- technical; added empty body virtual destructors to all struct/classes with a virtual methods 
- jenkins tests pass

**Comment by davidlange6** [comment]
merge

**Pull request merged**